pages/api/tool/[id]/module/post.ts

```diff
@@ -0,0 +1,33 @@
+import { NextApiRequest, NextApiResponse } from "next";
+import { Module } from "../../../models/module";
+import { Tool } from "../../../models/tool";
+
+const post = async (
+    req: NextApiRequest,
+    res: NextApiResponse,
+): Promise<void> => {
+    let postModule;
+    const { id } = req.query;
+    const tool = await Tool.findById(id).exec();
+
+    if (!tool)
+        return res
+            .status(404)
+            .send("The module with the given ID was not found.");
+
+    if (!req.body.title)
```

Comment:
You could call the post method here with the proper request params to not duplicate code (once the checks above are done)